node/loop.go

```diff
@@ -337,27 +356,40 @@ func (t *loopingTransport) loop() {
 	_ = t.conn.Close()
 }
 
-func (t *loopingTransport) nextID() jsonrpc.ID {
+func (t *loopingTransport) nextID(seed jsonrpc.ID) jsonrpc.ID {
+	n := atomic.AddUint64(&t.counter, 1)
+	if seed.IsString {
+		return jsonrpc.ID{
+			Num:      0,
+			Str:      fmt.Sprintf("%s-%d", seed.Str, n),
```

Comment:
I could be totally wrong, but would this cause a feedback loop?

Say we start with seed="foo", then we get "foo-1" for example. Next time we call it, wouldn't seed be "foo-1", so we'll get "foo-1-1"?

Reply:
I think it's ok, because we restore the original ID in the response at: 

https://github.com/INFURA/go-ethlibs/blob/master/node/loop.go#L182